lib/ts/recipe/emailpassword/api/implementation.ts

```diff
@@ -630,12 +631,23 @@ export default function getAPIImplementation(): APIInterface {
                 checkCredentialsOnTenant,
             });
 
+            const anomalyDetection = AnomalyDetectionRecipe.getInstance();
+
             const isVerified = authenticatingUser !== undefined && authenticatingUser.loginMethod!.verified;
             // We check this before preAuthChecks, because that function assumes that if isSignUp is false,
             // then authenticatingUser is defined. While it wouldn't technically cause any problems with
             // the implementation of that function, this way we can guarantee that either isSignInAllowed or
             // isSignUpAllowed will be called as expected.
             if (authenticatingUser === undefined) {
+                // we will intercept this request in favor to detect brute force attacks
+                if (anomalyDetection !== undefined) {
+                    await anomalyDetection.recipeInterfaceImpl.checkAnomalyWithRequest({
+                        request: options.req,
+                        status: "WRONG_CREDENTIALS_ERROR",
+                        tenantId,
+                        action: "SIGN_IN",
+                    });
```

Comment:
- we haven't consumed the result of this to block / report anything?
- Ideally we want to check for anomaly after the credentials have been verified? Not sure about this - what are your thoughts?

Reply:
the idea here is to have the ip / headers of a possible brute force attack.

Reply:
Maybe we should have two functions then:
- checkAnomaly
- registerInformation (or something like that). 

So it's very clear which one is what.

Reply:
will be cleaner for sure